### ISSUE TITLE
"setting will not be sownload from the server..." error message

### ISSUE THREAD

**Issue by Macfly** [mvberg/ib-gateway-docker#29]
Hello, I've been using this image for years but starting today, it stopped working and I have this error message.

Anyone know how to fix that?

![image](https://user-images.githubusercontent.com/1222059/165871498-9de7b6f0-d641-46be-93a4-30cc315d31a4.png)

**Comment by kukafree** [comment]
Now IBKR is blocking 974.4g: "The current version of the application you are running, 974.4g, is no longer supported. Please download the current version from our website."
I've got the repo file changes needed for switching to the IBC fork, currently supported by original IBController developer. Also upgraded to Gateway v1012. Let me know how to proceed: PR branch, fork, patch file, etc...

**Comment by krumware** [comment]
@kukafree is the fork you reference a for of the image or just the ib-controller itself. I have contributor access and can review a pull request.

@mvberg are you able to you weigh in?

**Comment by prodoramatlin** [comment]
> Now IBKR is blocking 974.4g: "The current version of the application you are running, 974.4g, is no longer supported. Please download the current version from our website." I've got the repo file changes needed for switching to the IBC fork, currently supported by original IBController developer. Also upgraded to Gateway v1012. Let me know how to proceed: PR branch, fork, patch file, etc...

Hi!

I'm REALLY stuck on this issue and i'm wondering if you could send me the updated version in some way? Please? :)

**Comment by Macfly** [comment]
> > Now IBKR is blocking 974.4g: "The current version of the application you are running, 974.4g, is no longer supported. Please download the current version from our website." I've got the repo file changes needed for switching to the IBC fork, currently supported by original IBController developer. Also upgraded to Gateway v1012. Let me know how to proceed: PR branch, fork, patch file, etc...
> 
> Hi!
> 
> I'm REALLY stuck on this issue and i'm wondering if you could send me the updated version in some way? Please? :)

I moved to another docket image and it is working great:
https://github.com/waytrade/ib-gateway-docker

**Comment by kukafree** [comment]
> I moved to another docket image and it is working great: https://github.com/waytrade/ib-gateway-docker

Funny, I guess I should have searched more. I spent a number of hours redoing what they had already done. I'll test out that image.

**Comment by kukafree** [comment]
> I moved to another docket image and it is working great: https://github.com/waytrade/ib-gateway-docker

Funny, I guess I should have searched more. I spent a number of hours redoing what they had already done. I'll test out that image.

**Comment by prodoramatlin** [comment]
> > > Now IBKR is blocking 974.4g: "The current version of the application you are running, 974.4g, is no longer supported. Please download the current version from our website." I've got the repo file changes needed for switching to the IBC fork, currently supported by original IBController developer. Also upgraded to Gateway v1012. Let me know how to proceed: PR branch, fork, patch file, etc...
> > 
> > 
> > Hi!
> > I'm REALLY stuck on this issue and i'm wondering if you could send me the updated version in some way? Please? :)
> 
> I moved to another docket image and it is working great: https://github.com/waytrade/ib-gateway-docker

Nice! Got it up and running after 10 mins. Nice work!

One thing though, I get this when trying to connect using my script:
Error 321, reqId -1: Error validating request.-'ca' : cause - The API interface is currently in Read-Only mode.

Where can I switch the Read-Only mode on/off?

**Comment by Macfly** [comment]
> > > > Now IBKR is blocking 974.4g: "The current version of the application you are running, 974.4g, is no longer supported. Please download the current version from our website." I've got the repo file changes needed for switching to the IBC fork, currently supported by original IBController developer. Also upgraded to Gateway v1012. Let me know how to proceed: PR branch, fork, patch file, etc...
> > > 
> > > 
> > > Hi!
> > > I'm REALLY stuck on this issue and i'm wondering if you could send me the updated version in some way? Please? :)
> > 
> > 
> > I moved to another docket image and it is working great: https://github.com/waytrade/ib-gateway-docker
> 
> Nice! Got it up and running after 10 mins. Nice work!
> 
> One thing though, I get this when trying to connect using my script: Error 321, reqId -1: Error validating request.-'ca' : cause - The API interface is currently in Read-Only mode.
> 
> Where can I switch the Read-Only mode on/off?

yes you need to VNC to the gateway and uncheck the read only mode.
to be able to vnc from another host thatn localhost host, you need to modify the docker-compose.yml and modify
- "127.0.0.1:5900:5900"
to
- "5900:5900"

**Comment by prodoramatlin** [comment]
> > > > > Now IBKR is blocking 974.4g: "The current version of the application you are running, 974.4g, is no longer supported. Please download the current version from our website." I've got the repo file changes needed for switching to the IBC fork, currently supported by original IBController developer. Also upgraded to Gateway v1012. Let me know how to proceed: PR branch, fork, patch file, etc...
> > > > 
> > > > 
> > > > Hi!
> > > > I'm REALLY stuck on this issue and i'm wondering if you could send me the updated version in some way? Please? :)
> > > 
> > > 
> > > I moved to another docket image and it is working great: https://github.com/waytrade/ib-gateway-docker
> > 
> > 
> > Nice! Got it up and running after 10 mins. Nice work!
> > One thing though, I get this when trying to connect using my script: Error 321, reqId -1: Error validating request.-'ca' : cause - The API interface is currently in Read-Only mode.
> > Where can I switch the Read-Only mode on/off?
> 
> yes you need to VNC to the gateway and uncheck the read only mode. to be able to vnc from another host thatn localhost host, you need to modify the docker-compose.yml and modify
> 
> * "127.0.0.1:5900:5900"
>   to
> * "5900:5900"

Thanx for the instructions! Just to clarify things, i'm shutting down my container every 24 hours. Does this mean that I need to VNC to the container every time I rebuild it?

**Comment by Macfly** [comment]
> > > > > > Now IBKR is blocking 974.4g: "The current version of the application you are running, 974.4g, is no longer supported. Please download the current version from our website." I've got the repo file changes needed for switching to the IBC fork, currently supported by original IBController developer. Also upgraded to Gateway v1012. Let me know how to proceed: PR branch, fork, patch file, etc...
> > > > > 
> > > > > 
> > > > > Hi!
> > > > > I'm REALLY stuck on this issue and i'm wondering if you could send me the updated version in some way? Please? :)
> > > > 
> > > > 
> > > > I moved to another docket image and it is working great: https://github.com/waytrade/ib-gateway-docker
> > > 
> > > 
> > > Nice! Got it up and running after 10 mins. Nice work!
> > > One thing though, I get this when trying to connect using my script: Error 321, reqId -1: Error validating request.-'ca' : cause - The API interface is currently in Read-Only mode.
> > > Where can I switch the Read-Only mode on/off?
> > 
> > 
> > yes you need to VNC to the gateway and uncheck the read only mode. to be able to vnc from another host thatn localhost host, you need to modify the docker-compose.yml and modify
> > 
> > * "127.0.0.1:5900:5900"
> >   to
> > * "5900:5900"
> 
> Thanx for the instructions! Just to clarify things, i'm shutting down my container every 24 hours. Does this mean that I need to VNC to the container every time I rebuild it?

not sure about that, according to the first post, feel like the settings are synchronized with the server.
why do you rebuild every day? you don't need this image with IB controller, only the official gateway is enough for you.

**Comment by prodoramatlin** [comment]
> > > > > > > Now IBKR is blocking 974.4g: "The current version of the application you are running, 974.4g, is no longer supported. Please download the current version from our website." I've got the repo file changes needed for switching to the IBC fork, currently supported by original IBController developer. Also upgraded to Gateway v1012. Let me know how to proceed: PR branch, fork, patch file, etc...
> > > > > > 
> > > > > > 
> > > > > > Hi!
> > > > > > I'm REALLY stuck on this issue and i'm wondering if you could send me the updated version in some way? Please? :)
> > > > > 
> > > > > 
> > > > > I moved to another docket image and it is working great: https://github.com/waytrade/ib-gateway-docker
> > > > 
> > > > 
> > > > Nice! Got it up and running after 10 mins. Nice work!
> > > > One thing though, I get this when trying to connect using my script: Error 321, reqId -1: Error validating request.-'ca' : cause - The API interface is currently in Read-Only mode.
> > > > Where can I switch the Read-Only mode on/off?
> > > 
> > > 
> > > yes you need to VNC to the gateway and uncheck the read only mode. to be able to vnc from another host thatn localhost host, you need to modify the docker-compose.yml and modify
> > > 
> > > * "127.0.0.1:5900:5900"
> > >   to
> > > * "5900:5900"
> > 
> > 
> > Thanx for the instructions! Just to clarify things, i'm shutting down my container every 24 hours. Does this mean that I need to VNC to the container every time I rebuild it?
> 
> not sure about that, according to the first post, feel like the settings are synchronized with the server. why do you rebuild every day? you don't need this image with IB controller, only the official gateway is enough for you.

Found this setting in config.ini:
ReadOnlyApi=

But i'm not sure how to override the config.ini from the package. Any clue?

**Comment by Macfly** [comment]
I think you have to modify this file:
https://github.com/waytrade/ib-gateway-docker/blob/master/config/ibgateway/jts.ini


Corentin Valleroy

**Comment by Neznakomec** [comment]
May be we should label, that this project is **deprecated**
due to the thing that IB doesn't allow version 974.4g as of now?
